typescript/cli/src/deploy/warp.ts

```diff
@@ -156,69 +158,54 @@ async function runBuildConfigStep({
   skipConfirmation: boolean;
 }) {
   log('Assembling token configs');
-  const { base, synthetics } = warpRouteConfig;
-  const { type: baseType, chainName: baseChainName, isNft } = base;
-
   const owner = await signer.getAddress();
-  const baseMetadata = await fetchBaseTokenMetadata(base, multiProvider);
-
-  log(
-    `Using base token metadata: Name: ${baseMetadata.name}, Symbol: ${baseMetadata.symbol}, Decimals: ${baseMetadata.decimals}`,
-  );
-
-  const mergedContractAddrs = getMergedContractAddresses(
-    coreArtifacts,
-    Object.keys(warpRouteConfig),
-  );
-
-  // Create configs that coalesce together values from the config file,
-  // the artifacts, and the SDK as a fallback
-  const configMap: ChainMap<TokenConfig & RouterConfig> = {
-    [baseChainName]: {
-      type: baseType,
-      token:
-        baseType === TokenType.collateral ||
-        baseType === TokenType.collateralVault
-          ? base.address!
-          : ethers.constants.AddressZero,
-      owner,
-      mailbox: base.mailbox || mergedContractAddrs[baseChainName]?.mailbox,
-      interchainSecurityModule:
-        base.interchainSecurityModule ||
-        mergedContractAddrs[baseChainName]?.interchainSecurityModule ||
-        mergedContractAddrs[baseChainName]?.multisigIsm,
-      // ismFactory: mergedContractAddrs[baseChainName].domainRoutingIsmFactory, // TODO fix when updating from routingIsm
-      foreignDeployment: base.foreignDeployment,
-      name: baseMetadata.name,
-      symbol: baseMetadata.symbol,
-      decimals: baseMetadata.decimals,
-    },
-  };
-
-  for (const synthetic of synthetics) {
-    const sChainName = synthetic.chainName;
-    configMap[sChainName] = {
-      type: TokenType.synthetic,
-      name: synthetic.name || baseMetadata.name,
-      symbol: synthetic.symbol || baseMetadata.symbol,
-      totalSupply: synthetic.totalSupply || 0,
-      owner,
-      mailbox: synthetic.mailbox || mergedContractAddrs[sChainName].mailbox,
-      interchainSecurityModule:
-        synthetic.interchainSecurityModule ||
-        mergedContractAddrs[sChainName]?.interchainSecurityModule ||
-        mergedContractAddrs[sChainName]?.multisigIsm,
-      // ismFactory: mergedContractAddrs[sChainName].domainRoutingIsmFactory, // TODO fix
-      foreignDeployment: synthetic.foreignDeployment,
-    };
-  }
-
-  // Request input for any address fields that are missing
   const requiredRouterFields: Array<keyof ConnectionClientConfig> = ['mailbox'];
-  let hasShownInfo = false;
-  for (const [chain, token] of Object.entries(configMap)) {
+  const remotes: string[] = [];
+
+  /// @dev This will keep track of the base collateral metadata which can get overwritten if there are multiple collaterals.
+  /// These 'base' variables are used to derive synthetic fields
+  /// @todo Remove this artifact when multi-collateral is enabled
```

Comment:
afaict this will still overwrite if there are multiple collaterals?

Reply:
Will resolve in separate PR